lib/canonical_version.sh

```diff
@@ -1,4 +1,28 @@
-canonicalVersion() {
+#!/usr/bin/env bash
+
+fetch_erlang_versions() {
+  url="https://raw.githubusercontent.com/HashNuke/heroku-buildpack-elixir-otp-builds/master/otp-versions"
+  curl -s "$url" 
+}
+
+exact_erlang_version_available() {
+  version=$1
+  available_versions=$2
+  found=1
+  while read -r line; do
+    if [ "$line" = "$version" ]; then
+      found=0
+    fi
+  done <<< "$available_versions"
+  echo $found
+}
+
+check_erlang_version() {
   version=$1
-  echo $version
+  exists=$(exact_erlang_version_available "$version" "$(fetch_erlang_versions)")
+  if [ $exists -ne 0 ]; then
```

Comment:
if you change `exact_erlang_version_available` to use `return` instead this can be written as:
```
if ! exact_erlang_version_available "$version" "$(fetch_erlang_versions)"; then

Reply:
Agreed, same as above.